Dockerfile

```diff
@@ -0,0 +1,71 @@
+FROM ubuntu:18.04 as env
+
+RUN apt-get update
+RUN apt-get install -y \
+    build-essential \
+    git \
+    libssl-dev \
+    lcov \
+    ruby \
+    wget
+
+# get an appropriate version of cmake
+RUN wget -qO- "https://cmake.org/files/v3.18/cmake-3.18.1-Linux-x86_64.tar.gz" | tar --strip-components=1 -xz -C /usr/local
+
+# set up certs for testing
+RUN mkdir /certs
```

Comment:
These can be scoped down to the build only for the `integtest`?

Reply:
That's a good idea. I was going to take a stab at getting demos working through actions now (many using certs) and I'm still working out the best way to keep these understandable while also reducing repetition.